ably/realtime_conn.go

```diff
@@ -118,6 +130,10 @@ func (c *Conn) connect(result bool) (Result, error) {
 	if err := c.auth.authQuery(query); err != nil {
 		return nil, c.setState(StateConnFailed, err)
 	}
+	if connKey != "" {
+		query.Set("resume", connKey)
+		query.Set("connectionSerial", fmt.Sprint(connSerial))
```

Comment:
Can you add comment here to mention that these are actually for `(RTN15b1)` and `(RTN15b2)`

Also we need tests for these, ensuring they are properly set.

Do you think maybe its a good idea to mention the spec id in tests so we can easily track down which tests have been covered.

Reply:
@gernest I actually meant to focus on [RTN15a](https://docs.ably.io/client-lib-development-guide/features/#RTN15a), which doesn't specify _how_ resume happens, just that it happens.

But anyway, it's true that the test wasn't really testing that the connection was recovered, just that there was a reconnection. 3404051 adds such to the test.

At 513b37f I've moved the test to a new file, where spec-oriented tests should be put. I did this already at #144, and I really intended this to be done for new API tests only, but since that's not happening yet, I think we should just do it now.

Reply:
👍